app/facades/cocktail_facade.rb

```diff
@@ -1,33 +1,31 @@
 class CocktailFacade
-  def self.retrieve_cocktail(cocktail_id)
+  def self.retrieve_cocktail(user_id, cocktail_id)
     response = CocktailService.get_cocktail_details(cocktail_id)
-
     details = response[:drinks].first
-
     ingredients = get_ingredients(details)
     measurements = get_measurements(details)
-
-    recipe = measurements.zip(ingredients).map do |i|
-      i.join if i != ["", ""]
-    end.compact
-
-    rating = if Rating.find_by(cocktail_id: cocktail_id).present?
-      Rating.find_by(cocktail_id: cocktail_id).stars
-    else
-      0
-    end
-
     {
-      # id: details[:idDrink],
       name: details[:strDrink],
       thumbnail: details[:strDrinkThumb],
       glass: details[:strGlass],
-      recipe: recipe,
+      recipe: build_recipe(ingredients, measurements),
       instructions: details[:strInstructions],
-      rating: rating
+      rating: find_rating(user_id, cocktail_id)
     }
+  end
 
-    # CocktailDetails.new(cocktail)
+  def self.find_rating(user_id, cocktail_id)
+    if Rating.find_by(user_id: user_id, cocktail_id: cocktail_id).present?
```

Comment:
small note... wonder if it looks cleaner to set this as a variable. as long as it works, works for me!

Reply:
I like that! Maybe could also be a method in the Rating model?

Reply:
Updated!